org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/TryWithResourcesFilter.java

```diff
@@ -0,0 +1,427 @@
+						&& nextIsVar(Opcodes.ALOAD, ctx + "t")
+						&& nextIsAddSuppressed() && nextIs(Opcodes.GOTO)
+						// "r.close()"
+						&& nextIsClose("r");
+			default:
+				return false;
+			}
+		}
+
+		private boolean matchEcj() {
+			// "catch (any primaryExc)"
+			if (!nextIsVar(Opcodes.ASTORE, "primaryExc")) {
+				return false;
+			}
+			if (!nextIsEcjCloseAndThrow("r0")) {
+				return false;
+			}
+
+			AbstractInsnNode c;
+			int resources = 1;
+			String r = "r" + resources;
+			c = cursor;
+			while (nextIsEcjClose(r)) {
+				if (!nextIsJump(Opcodes.GOTO, r + ".end")) {
+					return false;
+				}
+				if (!nextIsEcjSuppress(r)) {
+					return false;
+				}
+				if (!nextIsEcjCloseAndThrow(r)) {
+					return false;
+				}
+				resources++;
+				r = "r" + resources;
+				c = cursor;
+			}
+			cursor = c;
+			if (!nextIsEcjSuppress("last")) {
+				return false;
+			}
+			// "throw primaryExc"
+			if (!nextIsVar(Opcodes.ALOAD, "primaryExc")) {
+				return false;
+			}
+			if (!nextIs(Opcodes.ATHROW)) {
+				return false;
+			}
+
+			final AbstractInsnNode end = cursor;
+			c = start.getPrevious();
+			cursor = c;
+			while (!nextIsEcjClose("r0")) {
+				c = c.getPrevious();
+				cursor = c;
+				if (cursor == null) {
+					return false;
+				}
+			}
+			next();
+			if (cursor.getOpcode() != Opcodes.GOTO) {
+				return false;
+			}
+
+			output.ignore(c.getNext(), cursor);
+			output.ignore(start.getNext(), end);
+			return true;
+		}
+
+		private boolean matchEcjNoFlowOut() {
+			// "catch (any primaryExc)"
+			if (!nextIsVar(Opcodes.ASTORE, "primaryExc")) {
+				return false;
+			}
+
+			AbstractInsnNode c;
+			int resources = 0;
+			String r = "r" + resources;
+			c = cursor;
+			while (nextIsEcjCloseAndThrow(r) && nextIsEcjSuppress(r)) {
+				resources++;
+				r = "r" + resources;
+				c = cursor;
+			}
+			cursor = c;
+			// "throw primaryExc"
+			if (!nextIsVar(Opcodes.ALOAD, "primaryExc")) {
+				return false;
+			}
+			if (!nextIs(Opcodes.ATHROW)) {
+				return false;
+			}
+
+			final AbstractInsnNode end = cursor;
+			c = start.getPrevious();
+			cursor = c;
+			while (!nextIsEcjClose("r0")) {
+				c = c.getPrevious();
+				cursor = c;
+				if (cursor == null) {
+					return false;
+				}
+			}
+			for (int i = 1; i < resources; i++) {
+				if (!nextIsEcjClose("r" + i)) {
+					return false;
+				}
+			}
+
+			output.ignore(c.getNext(), cursor);
+			output.ignore(start, end);
+			return true;
+		}
+
+		private boolean nextIsEcjClose(final String name) {
+			return nextIsVar(Opcodes.ALOAD, name)
+					// "if (r != null)"
+					&& nextIsJump(Opcodes.IFNULL, name + ".end")
+					// "r.close()"
+					&& nextIsClose(name);
+		}
+
+		private boolean nextIsEcjCloseAndThrow(final String name) {
+			return nextIsVar(Opcodes.ALOAD, name)
+					// "if (r != null)"
+					&& nextIsJump(Opcodes.IFNULL, name)
+					// "r.close()"
+					&& nextIsClose(name) && nextIsLabel(name)
+					&& nextIs(Opcodes.ALOAD) && nextIs(Opcodes.ATHROW);
```

Comment:
this remaining unnamed `ALOAD` that was found in https://github.com/jacoco/jacoco/pull/500#discussion_r108958800 should be named "primaryExc"